__tests__/stopcovid/drill_progress/test_status.py

```diff
@@ -1,22 +1,40 @@
 import logging
 import unittest
+from unittest.mock import MagicMock
+from uuid import uuid4
 
 from stopcovid.dialog.models.events import (
     DrillStarted,
     UserValidated,
     NextDrillRequested,
     DialogEventBatch,
+    DrillCompleted,
 )
 from stopcovid.dialog.registration import CodeValidationPayload
-from stopcovid.dialog.models.state import UserProfile
+from stopcovid.dialog.models.state import UserProfile, DialogState
 from stopcovid.drills.drills import get_drill
-from stopcovid.drill_progress.status import initiates_first_drill, initiates_subsequent_drill
+from stopcovid.drill_progress.status import (
+    initiates_first_drill,
+    initiates_subsequent_drill,
+)
 
 
 class TestStatus(unittest.TestCase):
     def setUp(self) -> None:
         logging.disable(logging.CRITICAL)
-        self.drill = get_drill("01-sample-drill")
+        self.phone_number = "123456789"
+        self.first_drill = get_drill("01-sample-drill")
+        self.intake_drill = get_drill("00-intake")
+        self.repo = MagicMock()
+        self.repo_1 = MagicMock()
+        self.dialog_state = DialogState(
+            current_drill=self.intake_drill, phone_number=self.phone_number, seq="0"
+        )
+        self.repo.fetch_dialog_state = MagicMock(return_value=self.dialog_state)
+        self.dialog_state_1 = DialogState(
+            current_drill=self.first_drill, phone_number=self.phone_number, seq="0"
+        )
+        self.repo_1.fetch_dialog_state = MagicMock(return_value=self.dialog_state_1)
```

Comment:
It might be easier to instantiate a `TestDialogRepository(DialogRepository)` and implement fetch dialog state to return conditional dialog states by phone number

Reply:
Great idea. Implementing that now and will post a revision.